### PR TITLE
ci: Run rbac tests with unit on merge to main

### PR DESCRIPTION
**Pull request by gagantrivedi**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Create a workflow to run tests from private packages like rbac on merge to main 

## How did you test this code?

here: https://github.com/Flagsmith/flagsmith/actions/runs/6794365166

### CODE REVIEW

**Comment by gagantrivedi** [comment]
Depends on https://github.com/Flagsmith/flagsmith/pull/2685

**Comment by gagantrivedi** [comment]
Docs fix is here: https://github.com/Flagsmith/flagsmith/pull/2938

**Comment by khvn26** [comment]
I think this should modify the original `API Unit Tests` job, not add a new one.

**Comment by gagantrivedi** [comment]
> I think this should modify the original `API Unit Tests` job, not add a new one.

No, we also want to test core without any of these private packages(and don't want it to run on pull request)

File: .github/workflows/api-tests-with-private-packages.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      - name: integrate rbac tests
+        run: ln -s $GITHUB_WORKSPACE/flagsmith-rbac/tests tests/rbac_tests
+        shell: bash
+        working-directory: api
+
+      - name: Check for missing migrations
+        env:
+          DOTENV_OVERRIDE_FILE: .env-ci
+          opts: --no-input --dry-run --check
+        run: make django-make-migrations
```
Comment (matthewelwell) [inline .github/workflows/api-tests-with-private-packages.yml]: This should already have been done in the other repo, right?

File: .github/workflows/api-tests-with-private-packages.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        python-version: ['3.10', '3.11']
+
+    steps:
+      - name: Cloning repo
+        uses: actions/checkout@v3
+
+      - name: Create analytics database
+        env:
+          PGPASSWORD: postgres
+        run: createdb -h localhost -U postgres -p 5432 analytics
```
Comment (matthewelwell) [inline .github/workflows/api-tests-with-private-packages.yml]: Do we need the analytics DB for these tests?

**Comment by matthewelwell** [comment]
>>I think this should modify the original API Unit Tests job, not add a new one.
>
>No, we also want to test core without any of these private packages(and don't want it to run on pull request)

This is an interesting discussion, yeah. I guess theoretically, the unit tests should be isolated so we should be able to test them in a single workflow, right? The point regarding only running these on a merge to main (or ideally in the merge queue) is a good one though. 

@gagantrivedi can you update this to use the merge queue functionality?

**Comment by gagantrivedi** [comment]
> This is an interesting discussion, yeah. I guess theoretically, the unit tests should be isolated so we should be able to test them in a single workflow, right?

I guess the problem here is that we call all python tests unit tests(in the workflow atleast) maybe we should rename this workflow?

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**